lib/elixir/src/elixir_erl_for.erl

```diff
@@ -92,37 +98,46 @@ collect_filters([H | T], Acc) ->
 collect_filters([], Acc) ->
   {Acc, []}.
 
-build_inline(Ann, Clauses, Expr, Into, _Var, Acc, S) ->
+build_inline(Ann, Clauses, Expr, Into, Uniq, _Var, Acc, S) ->
   case lists:all(fun(Clause) -> element(1, Clause) == bin end, Clauses) of
```

Comment:
Just so we don't forget about it, we need to test this still:

```elixir
for <<x <- "abcabc">>, uniq: true, into: "", do: <<x>>
```

and that will require not going on th efirst clause if Uniq is set to true.